src/player/Player.php

```diff
@@ -832,6 +832,10 @@ public function doFirstSpawn() : void{
 		if($this->getHealth() <= 0){
 			$this->respawn();
 		}
+
+		foreach($this->server->getOnlinePlayers() as $player){
+			$player->getNetworkSession()->syncAvailableCommands();
+		}
```

Comment:
what is the purpose of this?

Reply:
PlayerParameter needs to synchronize every time the players join or leaves.

Reply:
But since the type is `target` the client will show a list of players for you. This is the same with the `Item` enum name, which will show a list of items available on the client (although BDS does actually send the entire item list for some reason, you don't actually need to)